### PR TITLE
Deprication of version R80.40 & R81 | AWS & AliCloud

### PR DESCRIPTION
**Pull request by almogard**

### CODE REVIEW

File: aws/templates/standalone/standalone.yaml [hunk 1]
```diff
@@ -456,66 +463,78 @@ Resources:
...
-  PublicAddress:
-    Type: AWS::EC2::EIP
-    Condition: AllocateAddress
-    Properties:
-      Domain: vpc
-  AddressAssoc:
-    Type: AWS::EC2::EIPAssociation
-    DependsOn: StandaloneInstance
-    Condition: AllocateAddress
+  GatewayLaunchTemplate:
```
Comment (chkp-noamcoh) [inline aws/templates/standalone/standalone.yaml:475]: there are a lot of changes here, what is the reason for that?

File: aws/templates/standalone/standalone-master.yaml [hunk 2]
```diff
@@ -157,6 +160,13 @@ Parameters:
...
+  MetaDataToken:
+    Description: Set true to deploy the instance with metadata v2 token required.
+    Type: String
+    Default: true
+    AllowedValues:
+      - true
+      - false
   StandaloneVersion:
     Description: Standalone Version & License.
     Type: String
```
Comment (chkp-noamcoh) [inline aws/templates/standalone/standalone-master.yaml:172]: deprecated versions are still here

File: aws/templates/tgw-ha/README.md [hunk 3]
```diff
@@ -23,4 +23,18 @@
...
 <br/>
-<br/>
\ No newline at end of file
+<br/>
+
+## Revision History
+In order to check the template version, please refer to [sk116585](https://supportcenter.checkpoint.com/supportcenter/portal?eventSubmit_doGoviewsolutiondetails=&solutionid=sk116585)
+
+| Template Version | Description                                                                                                      |
+|------------------|------------------------------------------------------------------------------------------------------------------|
```
Comment (chkp-noamcoh) [inline aws/templates/tgw-ha/README.md:32]: deprecation version is missing

File: terraform/alicloud/management-master/README.md [hunk 4]
```diff
@@ -121,10 +121,10 @@ bootstrap_script = "echo 'this is bootstrap script' > /home/admin/testfile.txt"
...
+| 20240704         | R81 version deprecation                                                                                                                                                                                            |
+| 20230829         | Change default Check Point version to R81.20                                                                                                                                                                       |
 | 20230615         | - Improved userdata quality and stability by moving to cloud-config<br/>- Define default primary and secondary NTP servers<br/>- Improved deployment experience for gateways and clusters managed by Smart-1 Cloud |
 | 20230521         | - Change default shell for the admin user to /etc/cli.sh<br/>- Add description for reserved words in hostname                                                                                                      |
-| 20230512         | New images with Jumbo Hotfix                                                                                                                                                                                       |
```
Comment (chkp-noamcoh) [inline terraform/alicloud/management-master/README.md:127]: why was this removed?

File: terraform/alicloud/management/README.md [hunk 5]
```diff
@@ -114,13 +114,13 @@ bootstrap_script = "echo 'this is bootstrap script' > /home/admin/testfile.txt"
...
 | 20230521         | - Change default shell for the admin user to /etc/cli.sh<br/>- Add description for reserved words in hostname                                                                                                      |
-| 20230512         | New images with Jumbo Hotfix                                                                                                                                                                                       |
 | 20230420         | Change alicloud terraform provider version to 1.203.0                                                                                                                                                              |
 | 20230330         | - Added support of ECS disk category. <br/> - Stability fixes.                                                                                                                                                     |
-| 20230129         | First release of R81.20 CloudGuard Management Terraform deployment in Alibaba Cloud.                                                                                                                               |
```
Comment (chkp-noamcoh) [inline terraform/alicloud/management/README.md:123]: why were the older versions changed/removed?

File: terraform/aws/cross-az-cluster/cluster_member_a_userdata.yaml [hunk 6]
```diff
@@ -1,4 +1,4 @@
...
\ No newline at end of file
+    python3 /etc/cloud_config.py enableCloudWatch=\"${EnableCloudWatch}\" sicKey=\"${SICKey}\" "smart1CloudToken=\"${TokenA}\"" installationType=\"cross-az-cluster\" osVersion=\"${OsVersion}\" allowUploadDownload=\"${AllowUploadDownload}\" templateVersion=\"20240310\" templateName=\"cross_az_cluster\" templateType=\"terraform\" shell=\"${Shell}\" enableInstanceConnect=\"${EnableInstanceConnect}\" hostName=\"${Hostname}\" ntpPrimary=\"${NTPPrimary}\" ntpSecondary=\"${NTPSecondary}\" passwordHash=\"${PasswordHash}\" MaintenanceModePassword=\"${MaintenanceModePassword}\" elasticIp=\"${MemberAPublicAddress}\" otherMemberIp=\"${MemberBPrivateAddressCluster}\" clusterIp=\"${PublicAddressCluster}\" secondaryIp=\"${MemberAPrivateAddressSecondary}\" otherMemberPrivateClusterIp=\"${MemberBPrivateAddressSecondary}\" bootstrapScript64=\"${GatewayBootstrapScript}\"
```
Comment (chkp-noamcoh) [inline terraform/aws/cross-az-cluster/cluster_member_a_userdata.yaml:4]: I think this is the wrong version number

File: terraform/aws/cross-az-cluster/cluster_member_b_userdata.yaml [hunk 7]
```diff
@@ -1,4 +1,4 @@
...
\ No newline at end of file
+    python3 /etc/cloud_config.py enableCloudWatch=\"${EnableCloudWatch}\" sicKey=\"${SICKey}\" "smart1CloudToken=\"${TokenB}\"" installationType=\"cross-az-cluster\" osVersion=\"${OsVersion}\" allowUploadDownload=\"${AllowUploadDownload}\" templateVersion=\"20240310\" templateName=\"cross_az_cluster\" templateType=\"terraform\" shell=\"${Shell}\" enableInstanceConnect=\"${EnableInstanceConnect}\" hostName=\"${Hostname}\" ntpPrimary=\"${NTPPrimary}\" ntpSecondary=\"${NTPSecondary}\" passwordHash=\"${PasswordHash}\" MaintenanceModePassword=\"${MaintenanceModePassword}\" elasticIp=\"${MemberBPublicAddress}\" otherMemberIp=\"${MemberAPrivateAddressCluster}\" clusterIp=\"${PublicAddressCluster}\" secondaryIp=\"${MemberBPrivateAddressSecondary}\" otherMemberPrivateClusterIp=\"${MemberAPrivateAddressSecondary}\" bootstrapScript64=\"${GatewayBootstrapScript}\"
```
Comment (chkp-noamcoh) [inline terraform/aws/cross-az-cluster/cluster_member_b_userdata.yaml:4]: I think this is the wrong version number

File: terraform/aws/gwlb/README.md [hunk 8]
```diff
@@ -208,20 +208,22 @@ In order to check the template version, please refer to [sk116585](https://suppo
...
-| 20221226         | Support ASG Launch Template instead of Launch Configuration                           |
-| 20230521         | Change default shell for the admin user to /etc/cli.sh                                |
-| 20230806         | Add support for c6in instance type                                                    | 
-| 20230829         | Change default Check Point version to R81.20                                          |
-| 20230910         | Add bootstrap script execution option for deployed gateways                           |
-| 20230914         | Add support for maintenance mode password                                             |
-| 20230923         | Add support for C5d instance type                                                     |
-| 20231012         | Update AWS Terraform provider version to 5.20.1                                       |
-| 20231022         | Fixed template to populate x-chkp-tags correctly                                      |
+| 20240704         | R80.40 version deprecation                                                            |
```
Comment (chkp-noamcoh) [inline terraform/aws/gwlb/README.md:211]: looks like R81 was removed as well

**Review by chkp-noamcoh — COMMENTED** [review]
There are a lot of empty files, what is the reason for that?
Another thing, the permissions of the CFTs README files been changed, why is that?

File: aws/templates/cluster/cluster.yaml [hunk 9]
```diff
@@ -537,49 +546,17 @@ Resources:
             - - !Join [ '=', [ cluster-ip, !Ref ClusterPublicAddress ] ]
               - !Join [ '=', [ cluster-eth0-private-ip, !Select [ 0, !GetAtt MemberAExternalInterface.SecondaryPrivateIpAddresses ] ] ]
               - !Join [ '=', [ cluster-eth1-private-ip, !Select [ 0, !GetAtt MemberAInternalInterface.SecondaryPrivateIpAddresses ] ] ]
-      ImageId: !GetAtt AMI.Outputs.ImageId
```
[the remark this answers is not in the document]
Comment (almogard) [reply aws/templates/cluster/cluster.yaml:540]: because I added the MetaDataToken update to that PR

[the remark this answers is not in the document]
Comment (almogard) [reply aws/templates/cluster/cluster.yaml:540]: and which files are empty? there are files that I only move to other folder to organized, so the "change" is only on the name of the file.

File: aws/templates/gwlb-asg/tgw-gwlb.yaml [hunk 10]
```diff
@@ -506,6 +520,11 @@ Parameters:
     AllowedValues:
       - true
       - false
+  GatewayBootstrapScript:
```
[the remark this answers is not in the document]
Comment (almogard) [reply aws/templates/gwlb-asg/tgw-gwlb.yaml:523]: I saw a different between gitlab and github, and that change has already been in others templates, only in that template its missing.

File: terraform/aws/cross-az-cluster/cluster_member_a_userdata.yaml (hunk 6 above)
Comment (almogard) [reply terraform/aws/cross-az-cluster/cluster_member_a_userdata.yaml:4]: I think its was like this before, I didn't change something in that template.

File: terraform/aws/cross-az-cluster/cluster_member_b_userdata.yaml (hunk 7 above)
Comment (almogard) [reply terraform/aws/cross-az-cluster/cluster_member_b_userdata.yaml:4]: same as above

File: terraform/aws/gwlb/README.md (hunk 8 above)
Comment (almogard) [reply terraform/aws/gwlb/README.md:211]: GWLB not supported on R81 & R81.10

File: terraform/aws/tgw-gwlb-master/README.md [hunk 11]
```diff
@@ -247,17 +246,18 @@ In order to check the template version, please refer to [sk116585](https://suppo
...
-| 20221226         | Support ASG Launch Template instead of Launch Configuration                                                                |
-| 20230806         | Add support for c6in instance type                                                                                         | 
-| 20230829         | Change default Check Point version to R81.20                                                                               |
-| 20230910         | Add bootstrap script execution option for deployed gateways                                                                |
-| 20230914         | Add support for maintenance mode password                                                                                  |
-| 20230923         | Add support for C5d instance type                                                                                          |
-| 20231012         | Update AWS Terraform provider version to 5.20.1                                                                            |
+| 20240704         | R80.40 version deprecation                                                                                                 |
```
[the remark this answers is not in the document]
Comment (almogard) [reply terraform/aws/tgw-gwlb-master/README.md:249]: fix that, but I don't know how to add that comment to this pull request

File: aws/templates/cluster/cluster.yaml (hunk 9 above)
[the remark this answers is not in the document]
Comment (chkp-noamcoh) [reply aws/templates/cluster/cluster.yaml:540]: for example:
image

File: aws/templates/gwlb-asg/tgw-gwlb.yaml (hunk 10 above)
[the remark this answers is not in the document]
Comment (chkp-noamcoh) [reply aws/templates/gwlb-asg/tgw-gwlb.yaml:523]: is it included in the same template in gitlab?

**Review by chkp-noamcoh — APPROVED** [review]
